crates/matrix-sdk/src/widget/machine/mod.rs

```diff
@@ -616,10 +651,94 @@ impl WidgetMachine {
                 let subscribe_required = !approved.read.is_empty();
                 machine.capabilities = CapabilitiesState::Negotiated(approved.clone());
 
+                let state_filters: Vec<_> = approved
+                    .read
+                    .clone()
+                    .into_iter()
+                    .filter_map(|f| match f {
+                        EventFilter::State(f) => Some(f),
+                        _ => None,
+                    })
+                    .collect();
+
                 let update = NotifyCapabilitiesChanged { approved, requested };
-                let (_request, action) = machine.send_to_widget_request(update);
+                let (_request, notify_action) = machine.send_to_widget_request(update);
+
+                let (initial_state_pending, guard) = {
+                    let token = CancellationToken::new();
+                    (token.child_token(), token.drop_guard())
+                };
+
+                if !state_filters.is_empty() {
+                    machine.pending_state_updates = Some(InitialStateUpdate {
```

Comment:
Are we sure there's no previous `machine.pending_state_updates` here? If so, can you add an `assert!()` checking for this? Or a code comment explaining why it's safe to override, otherwise?